# 02 - Randomised Experiments

## The Golden Standard

1장에서는 상관관계가 어떻게 인과관계와 다른지 알아봤습니다. 또한, 상관관계를 인과관계로 만드는 데 필요한 것을 살펴보았습니다.

$
E[Y|T=1] - E[Y|T=0] = \underbrace{E[Y_1 - Y_0|T=1]}_{ATT} + \underbrace{\{ E[Y_0|T=1] - E[Y_0|T=0] \}}_{BIAS}
$

요약하면 편향이 없다면 상관관계는 인과관계입니다. $E[Y_0|T=0]=E[Y_0|T=1]$일 때, 편향이 없습니다. 즉, 처치에 의한 효과를 제거한 실험군이 대조군과 같거나 비교 가능하면 상관관계는 인과관계가 됩니다. 기술적으로 `untreated outcome`(처치되지 않은 결과)가 `treated outcome`(처치된 결과)의 `counterfactual outcome`(반사실적 결과)와 같습니다. '반사실적 결과'는 '처치된 그룹이 처치를 받지 못했을 경우 얻었을 가상의 결과'라는 점을 기억하세요.

상관관계를 수학적으로 인과관계와 같게 만드는 방법을 알아봤지만 이론일 뿐입니다. 편향을 없애기 위한 첫 번째 도구인 **Randomized Experiments**(무작위 실험)를 살펴보겠습니다. 무작위 실험은 개별 데이터를 실험군이나 대조군에 무작위로 할당하는 방법입니다. 실험군 비율이 50%일 필요는 없습니다. 원한다면 실험군을 전체의 10%로 설정할 수도 있습니다.

무작위화는 `potential outcome`(잠재적 결과)를 `treatment`와 독립적으로 만들고 편향을 없앱니다.

$
(Y_0, Y_1) \perp\!\!\!\perp T
$

처음에는 헷갈릴 수 있어요. (저는 그랬습니다.) 걱정 마시오. 나의 진정한 친구. 자세히 설명 드리겠습니다. 만약 `outcome`이 `treatment`와 독립이라면, `treatment` 효과가 없다는 것을 의미하지 않을까요? 맞습니다! 하지만 우리는 `outcome`에 대해 말하는 것이 아닙니다. **potential outcome**을 이야기하고 있어요. `potential outcome`은 `outcome`이 어떻게 처치($Y_1$) 또는 통제($Y_0$) **받았을까**에 대한 것입니다. 무작위 실험에서 우리는 **처치가 결과를 바꾼다고** 생각하기 때문에 결과와 처치가 **독립적일 것으로 원하지 않습니다.** 그러나 **potential outcome**은 `treatment`와 독립이길 원합니다.

<figure>
    <img src="data/img/rct/indep.png">
    <figcaption>원하는 것과 아닌 것</figcaption>
</figure>

`potential outcome`이 `treatment`와 독립이라는 말은 실험군과 대조군이 동일함을 의미합니다. 쉽게 말해 실험군과 대조군은 비교 가능합니다. `treatment`의 할당 여부를 아는 것은 처치 전 `outcome`에 대한 어떠한 정보도 제공하지 않습니다. 결과적으로 $(Y_0, Y_1)\perp T$는 오직 `treatment` 만이 실험군과 대조군 사이의 차이라는 것을 말해줍니다. 독립성은 정확하게 아래 수식과 같은 것에 주목하세요.

$
E[Y_0|T=0]=E[Y_0|T=1]=E[Y_0]
$

1장에서 본 것처럼 아래 수식을 유도할 수 있습니다.

$
E[Y|T=1] - E[Y|T=0] = E[Y_1 - Y_0]=ATE
$

무작위화는 실험군과 대조군 사이의 평균 차이를 사용하는 방법을 제공하며 이를 `treatment effect`라 부릅니다.


## In a School Far, Far Away

2020년 COVID-19는 기업이 사회적 거리두기에 적응하도록 강요했습니다. 배달 서비스가 널리 보급되고 대기업은 재택근무로 전환했습니다. 학교도 다르지 않았습니다. 많은 학교가 온라인 강의를 시작했습니다.

위기가 발생한 지 4개월이 지난 지금, 사람들은 변화가 유지될지 궁금해합니다. 온라인 학습의 이점은 의심할 여지가 없습니다. 온라인 학습은 월세나 교통비를 절약할 수 있어 저렴합니다. 또한, 전 세계 세계적인 수준의 콘텐츠로 더 디지털화할 수 있습니다. 하지만 여전히 우리는 온라인 학습이 학생들의 학업 성취에 부정적인 영향을 미치는지 긍정적인 영향을 미치는지 대답할 필요가 있습니다.

답변 중 하나는 온라인 수업을 하는 학교의 학생의 성적과 전통적인 교실에서 강의하는 학교의 학생의 성적을 비교하는 것입니다. 하지만 이는 최선의 접근법이 아닙니다. 온라인 학교는 잘 훈련된 학생만 끌어들일 수 있습니다. 이 경우, `treatment`(온라인 강의)가 적용될 때 양의 편향이 있을 것입니다: $E[Y_0|T=1] > E[Y_0|T=0]$.

반면 온라인 수업이 저렴하므로 공부 외에 일을 하는 사람들로 구성될 수 있습니다. 온라인 수업 학생들은 더 나쁜 성적을 낼 것이고, `treatment`(온라인 강의)가 적용될 때 음의 편향이 있을 것입니다: $E[Y_0|T=1] < E[Y_0|T=0]$.

간단한 비교는 설득력이 없습니다. 인과영향을 감추는 편향이 없는지 확신할 수 없었습니다.

![img](./data/img/rct/lurking_bias.png)

해결을 위해 비교 가능한 실험군과 대조군이 필요합니다. $E[Y_0|T=1] = E[Y_0|T=0]$. 한 가지 방법은 온라인 수업을 학생들에게 무작위로 할당하는 것입니다. 이렇게 할 수 있다면 `treatment` 외에는 평균적으로 같을 것입니다.

다행히 몇몇 연구 결과가 있습니다. 그들은 무작위로 일부 학생은 대면 수업을, 일부 학생들은 온라인 수업을, 나머지 학생은 대면과 온라인이 혼합된 수업이 진행되도록 했습니다. 또한, 학기 말 시험 데이터를 수집했습니다.

데이터는 아래와 같습니다.

In [1]:
import pandas as pd
import numpy as np

data = pd.read_csv("./data/online_classroom.csv")
print(data.shape)
data.head()

(323, 10)


gender  asian  black  hawaiian  hispanic  unknown  white  format_ol  \
0       0    0.0    0.0       0.0       0.0      0.0    1.0          0   
1       1    0.0    0.0       0.0       0.0      0.0    1.0          0   
2       1    0.0    0.0       0.0       0.0      0.0    1.0          0   
3       1    0.0    0.0       0.0       0.0      0.0    1.0          0   
4       1    0.0    0.0       0.0       0.0      0.0    1.0          1   

   format_blended  falsexam  
0             0.0  63.29997  
1             0.0  79.96000  
2             1.0  83.37000  
3             1.0  90.01994  
4             0.0  83.30000

323개 데이터가 포함되어 있습니다. 빅데이터는 아니지만 충분합니다. 인과 관계를 추정하기 위해 각 그룹의 평균 점수를 간단히 계산할 수 있습니다.

In [2]:
(data
 .assign(class_format = np.select(
     [data["format_ol"].astype(bool), data["format_blended"].astype(bool)],
     ["online", "blended"],
     default="face_to_face"
 ))
 .groupby(["class_format"])
 .mean())

gender     asian     black  hawaiian  hispanic   unknown  \
class_format                                                               
blended       0.550459  0.217949  0.102564  0.025641  0.012821  0.012821   
face_to_face  0.633333  0.202020  0.070707  0.000000  0.010101  0.000000   
online        0.542553  0.228571  0.028571  0.014286  0.028571  0.000000   

                 white  format_ol  format_blended   falsexam  
class_format                                                  
blended       0.628205        0.0             1.0  77.093731  
face_to_face  0.717172        0.0             0.0  78.547485  
online        0.700000        1.0             0.0  73.635263

간단하죠? 대면 수업은 평균 점수가 78.54점이지만 온라인 수업은 73.63점임을 알 수 있습니다. 온라인 학습 지지자에게는 좋은 소식이 아니네요. 온라인 학습의 $ATE$는 -4.91입니다. 이는 **온라인 수업이 평균 점수를 약 5점 낮게함**을 의미합니다. 온라인 강의가 대면 수업을 할 수 없는 가난한 학생들로 구성되어 있다고 걱정할 필요가 없습니다. 다른 처치를 받은 학생이 `treatment` 외에 어떤 면에서도 다르다는 것을 걱정할 필요가 없습니다. 무작위 실험은 `treatment` 외 차이를 제거하기 위해 설계되었습니다.

무작위화가 제대로 수행되었는지 (올바른 데이터를 보고 있는지) 확인하는 방법은 `treatment`된 데이터가 그렇지 않은 데이터와 동일한지 확인하는 것입니다. 데이터는 성별과 민족 정보를 가지고 있으므로 그룹 간 유사 여부를 확인할 수 있습니다. `gender`, `asian`, `hispanic`, `white` 변수는 상당히 유사하다 할 수 있습니다. 하지만 `black` 변수는 조금 다른 것 같습니다. 우리는 작은 데이터 세트에서 어떤 일이 일어나는지 관심을 둬야 합니다. 무작위화를 하더라도 우연히 한 그룹이 다른 그룹과 다를 수 있습니다. 큰 데이터 세트에서는 이러한 차이가 줄어드는 경향이 있습니다.

## The Ideal Experiment

`Randomised experiments`(무작위실험)이나 `Randomised Controlled Trials`(RCT, 무작위임상실험)은 인과효과를 얻을 수 있는 가장 신뢰할 수 있는 방법입니다. 직접적인 기술이며 너무나 설득력 있습니다. 매우 강력하여 대부분 국가에서 신약 효과를 증명하기 위한 요구사항으로 채택됩니다. 가능하다면 RCT는 인과관계를 밝혀내기 위해 할 수 있는 모든 것이 될 것입니다. 잘 설계된 RCT는 모든 과학자의 꿈입니다.

![img](./data/img/rct/science_dream.png)

안타깝게도 RCT는 비용이 매우 많이 들거나 비윤리적일 수 있습니다. 종종 단순히 할 수 없기도 합니다. 임신 중 흡연이 아기 체중에 미치는 영향을 추정하려는 의사를 상상해 봅시다. 의사는 무작위로 임신 중인 엄마들에게 담배를 피우라고 강요는 할 수 없습니다. 여러분이 큰 은행에서 일하고 있고, 신용 한도가 고객 이탈에 미치는 영향을 알고 싶다고 해봅시다. 고객에게 무작위로 신용한도를 설정하는 것은 너무 큰 비용이 들 것입니다. 최저임금 인상이 실업률에 미치는 영향을 이해하고 싶나요? 국가들에 무작위로 최저임금을 할당하라 할 수는 없습니다.

나중에 조건부 무작위화로 비용을 낮추는 방법을 볼 예정이지만, 비윤리적이거나 실현 불가능한 실험에 대해서는 할 수 있는 것이 없습니다. 그럼에도 인과적 질문에 대한 **이상적인 실험**을 생각해 볼 가치는 있습니다. **인과 효과를 밝히기 위해 할 수 있는 완벽한 실험이 무엇인지 항상 생각해 보세요.** 이상적인 실험 없이도 인과 효과를 발견할 방법을 알려줄지 모릅니다.

## The Assignment Mechanism

무작위 실험에서 `unit`에 `treatment`를 할당하는 방법은 무작위입니다. 앞으로 등장하는 모든 인과 추론 기법은 `treatment`의 할당 방법을 식별하려 할 것입니다. 할당 방법이 어떻게 동작하는지 알게 되면, 무작위가 아니어도 인과 추론에 자신감이 붙을 겁니다.

할당 메커니즘은 데이터만으로 찾을 수 없습니다. 예를 들어 교육과 재산에 관련된 데이터 세트를 가지고 있다면, 데이터만으로는 어떤 것이 원인인지 알 방법은 없습니다. 당신은 세상이 어떻게 돌아가는지에 대한 배경지식으로 그럴듯한 방법을 주장해야 할 것입니다. 학교가 사람을 교육하고, 더 생산적으로 만들고, 더 높은 임금을 받는 직업으로 이끈다는 것이 사실일까요? 만약 당신이 교육에 대해 비관적인 사람이라면, 학교는 생산성을 높이기 위해 아무것도 하지 않는다고 말할 수 있습니다. 단지 부유한 가정의 아이가 더 높은 학위를 받는 거짓된 상관관계이기 때문입니다.

인과질문에서 X가 Y를 유발한다고 주장할 수 있습니다. 또는, X와 Y를 모두 유발하는 세 번째 변수 Z가 존재하므로 X와 Y 상관관계는 거짓이라 주장할 수도 있습니다. 따라서 할당 매커니즘을 아는 것은 훨씬 설득력 있는 답변으로 이어집니다. 이러한 특징은 인과 추론을 매우 흥미롭게 만듭니다. ML은 단순히 적절한 순서로 일부 버튼을 누르는 것에 불과하지만, 인과 추론은 데이터 생성 메커니즘을 진지하게 생각해야 합니다.

## Key Ideas

인과성을 발견하는 가장 간단하고 효과적인 방법인 무작위 실험에 대해 살펴봤습니다. 무작위 실험은 실험군과 대조군을 비교할 수 있게 만들어 줍니다. 항상 무작위 실험을 할 수는 없지만 이상적인 실험에 대해 생각하는 것은 도움이 될 것입니다.

통계에 익숙한 사람은 인과 관계 추정치를 평가하지 않았다고 지적할 수 있어요. 4.91점 하락이 우연이 아니라는 것을 어떻게 알 수 있나요? 차이가 통계적으로 유의한지 어떻게 알 수 있습니까? 타당한 의견입니다. 걱정하지 마세요. 다음 장에서 몇 가지 통계적 개념을 살펴보겠습니다.

## References

I like to think of this entire book as a tribute to Joshua Angrist, Alberto Abadie and Christopher Walters for their amazing Econometrics class. Most of the ideas here are taken from their classes at the American Economic Association. Watching them is what is keeping me sane during this tough year of 2020.
* [Cross-Section Econometrics](https://www.aeaweb.org/conference/cont-ed/2017-webcasts)
* [Mastering Mostly Harmless Econometrics](https://www.aeaweb.org/conference/cont-ed/2020-webcasts)

I'll also like to reference the amazing books from Angrist. They have shown me that Econometrics, or 'Metrics as they call it, is not only extremely useful but also profoundly fun.

* [Mostly Harmless Econometrics](https://www.mostlyharmlesseconometrics.com/)
* [Mastering 'Metrics](https://www.masteringmetrics.com/)

My final reference is Miguel Hernan and Jamie Robins' book. It has been my trustworthy companion in the most thorny causal questions I had to answer.

* [Causal Inference Book](https://www.hsph.harvard.edu/miguel-hernan/causal-inference-book/)

The data used here is from a study of Alpert, William T., Kenneth A. Couch, and Oskar R. Harmon. 2016. ["A Randomized Assessment of Online Learning"](https://www.aeaweb.org/articles?id=10.1257/aer.p20161057). American Economic Review, 106 (5): 378-82.

![img](./data/img/poetry.png)

## Contribute

Causal Inference for the Brave and True is an open-source material on causal inference, the statistics of science. Its goal is to be accessible monetarily and intellectually. It uses only free software based on Python.
If you found this book valuable and want to support it, please go to [Patreon](https://www.patreon.com/causal_inference_for_the_brave_and_true). If you are not ready to contribute financially, you can also help by fixing typos, suggesting edits, or giving feedback on passages you didn't understand. Go to the book's repository and [open an issue](https://github.com/matheusfacure/python-causality-handbook/issues). Finally, if you liked this content, please share it with others who might find it helpful and give it a [star on GitHub](https://github.com/matheusfacure/python-causality-handbook/stargazers).